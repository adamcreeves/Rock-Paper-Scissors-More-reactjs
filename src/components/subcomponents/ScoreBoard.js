import React from "react";
import "../../styles/Main.css";
import Title from "./Title";

function ScoreBoard({ playerScore, opponentScore }) {
  return (
    <div className="scoreBoard">
      <Title str={"Score:"} classNm={"scoreText"} />
      <div className="scoreBoard__body">
        <div className="scoreBoard__row">
          <div className="scoreBoard__label">{"User: "}</div>
          <div className="scoreBoard__score">{playerScore}</div>
        </div>
        <div className="scoreBoard__row">
          <div className="scoreBoard__label">{"Computer: "}</div>
          <div className="scoreBoard__score">{opponentScore}</div>
        </div>
      </div>
    </div>
  );
}

export default ScoreBoard;
