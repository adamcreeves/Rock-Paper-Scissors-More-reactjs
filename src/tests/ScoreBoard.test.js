import React from "react";
import { render } from "@testing-library/react";
import ScoreBoard from "../components/subcomponents/ScoreBoard";

describe("Test Body", () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(<ScoreBoard />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render Body and not be null", () => {
    const { asFragment } = render(<ScoreBoard />);
    expect(asFragment()).not.toBeNull();
  });
});
