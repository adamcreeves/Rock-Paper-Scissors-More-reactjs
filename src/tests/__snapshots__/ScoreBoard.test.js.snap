// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Body should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="scoreBoard"
  >
    <div
      class="scoreBoard__title"
      data-testid="title-component"
    >
      Score:
    </div>
    <div
      class="scoreBoard__body"
    >
      <div
        class="scoreBoard__row"
      >
        <div
          class="scoreBoard__label"
        >
          undefined: 
        </div>
        <div
          class="scoreBoard__score"
        />
      </div>
      <div
        class="scoreBoard__row"
      >
        <div
          class="scoreBoard__label"
        >
          undefined: 
        </div>
        <div
          class="scoreBoard__score"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
